Objects/frameobject.c

```diff
@@ -206,6 +206,7 @@ mark_stacks(PyCodeObject *code_obj, int len)
                 case POP_JUMP_IF_FALSE:
                 case POP_JUMP_IF_TRUE:
                 case JUMP_IF_NOT_EXC_MATCH:
+                case JUMP_IF_NOT_EG_MATCH:
```

Comment:
I'll let Mark decide whether this is all you need here -- to me, it looks a little suspicious that you treat 'EG' the same as 'EXC' but there's no check for 'EG' below around line 217.

Reply:
I agree, this looks wrong.

Reply:
I don't know that I fixed it, but I made it look less wrong. @markshannon what is this function doing?

Reply:
I tried removing both JUMP_IF_NOT_EXC_MATCH and JUMP_IF_NOT_EG_MATCH from this function to see what tests will break, but none did.

If I comment out the whole block except the 'break' (which impacts all the jumps) then some trace tests break fail (test_sys_settrace.py).

Reply:
I spoke with @markshannon offline. This function’s job is to make sure that jumping to some line in the debugger is safe. So if you ask to jump to the middle of an except block it handles that. It is not well covered by tests, and if it’s wrong this means there is a bug in some edge case in the debugger. So we can worry about it separately.